### PR TITLE
added functions for performing generalized kendal tau 

### PR DESCRIPTION
**Pull request by ysong-astro**
partially vectorized bootstrap and perturbation procedures, but code still runs pretty slow because kendall function is not vectorized, hence still requires iterating to calculate each pair of tau and pvalue.

### CODE REVIEW

File: pymcspearman/pymckendall.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    pval=st.norm.sf(abs(z))*2
+    return tau,pval
+
+
+def pymckendall(x, y, xlim, ylim, dx=None, dy=None, Nboot=10000, Nperturb=10000,
+    bootstrap=True,
+    perturb=True,
+    percentiles=(16,50,84), return_dist=False):
+    """
+    Compute spearman rank coefficient with uncertainties using several methods.
```
Comment (privong) [inline pymcspearman/pymckendall.py]: minor: this should be changed to say Kendall tau

File: pymcspearman/pymckendall.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+pymckendall.py
+
+Python implementation of Isobe, Feigelson & Nelson (1986) method for calculating Kendall
+tau rank correlation coefficient with uncertainties on censored data (upper/lowerlimit).
+
+"""
+
+import numpy as np
+from matplotlib import pyplot as plt
+import pandas as pd
```
Comment (privong) [inline pymcspearman/pymckendall.py]: I don't think pandas is used, so you can remove this import?

Comment (privong) [reply pymcspearman/pymckendall.py]: Same with the matplotlib import

File: pymcspearman/pymckendall.py (hunk 1 above)
Comment (ysong-astro) [reply pymcspearman/pymckendall.py]: changed

File: pymcspearman/pymckendall.py (hunk 2 above)
Comment (ysong-astro) [reply pymcspearman/pymckendall.py]: removed unused packages

**Review by privong — APPROVED** [review]

**Pull request merged**